### ISSUE TITLE
Add Nostr Web Provider

### ISSUE THREAD

**Issue by tiero** [vulpemventures/marina#279]
Would be useful to add support for Nostr protocol events, used to build alternative decentralized Twitter and so on

https://github.com/fiatjaf/nostr

```js
window.nostr.getPublicKey
window.nostr.signEvent
```

Web provider: https://github.com/fiatjaf/nos2x/blob/master/extension/nostr-provider.js

**Comment by bordalix** [comment]
@tiero can you describe a use case where this would be useful for/with Marina?

I'm not sure if you mean:
a) allow Marina to be used also as a nostr extension (inject `window.nostr`)
b) or to use `nostr.signEvent` inside Marina for some new functionality

**Comment by tiero** [comment]
> a) allow Marina to be used also as a nostr extension (inject `window.nostr`) 

yes, to be able to use the Marina to inject a nostr web provider. So it will be injected in parallel with the `marina` one. 

Ideally we should derive a key pair from the xpub and use it as "permanent identity" public Key
